### DISCUSSION TITLE
sensor.community yaml does not install

### DISCUSSION THREAD

**Discussion by pe0ter** [stas-sl/esphome-sound-level-meter#22] [Q&A] [answered]
Sorry to bother, I am fairly new to EspHome and have little skill in programming. I am running EspHome on a Home Assistant Yellow.
My aim is building a small and cheap dnms device with solid measurement quality and good wifi.  People in my town are asking for this to record their noise environments. I have selected a seeed xiao ESP32-S3 with PSRAM and a ICS43434 mic for this and want to send measurement data to sensor.community for general logging and make this data available to others like our RIVM and city council.
Searching for the right software for this I found your topic.

I copied and tried to install your sensor-community-example-config.yaml but I get compilation errors pointing to wdt functions/files not found.

Could you point me in the right direction to get your yaml installed?

Best Regards,
Peter

------------------------------------------------------------
substitutions:
  name: esphome-web-81d65c
  friendly_name: Xiao ESP32-S3 ICS43434 DNMS

external_components:
  - source: github://stas-sl/esphome-sound-level-meter

esphome:
  name: ${name}
  friendly_name: ${friendly_name}
  name_add_mac_suffix: false

  includes:
    # should contain single line: #include <esp_task_wdt.h>
    - wdt_include.h                                                              <------ file not found error
    
esp_task_wdt_init()
    
  platformio_options:
    board_build.flash_mode: dio
    build_flags: "-DBOARD_HAS_PSRAM"
    board_build.arduino.memory_type: qio_opi
  project:
    name: esphome.web
    version: '1.0'
  on_boot:
    then:
      - lambda: !lambda |-                                            <---- when I comment out the wdt_include.h script starts compiling but stops here
          // increase watchdog timeout to 30 seconds
          // so that ESP32 don't crash on long http requests
          // which might happen with sensor.community quite often
          esp_task_wdt_init(30, false);

esp32:
  board: esp32-s3-devkitc-1
  framework:
    type: arduino

psram:

# Enable logging
logger:
  level: DEBUG

**Comment by stas-sl** [comment, marked as answer]
You have to put (create) `wdt_include.h` file in the same folder as your config. It should contain only a single line:
```cpp
#include <esp_task_wdt.h>
```
Or you can just copy it from https://github.com/stas-sl/esphome-sound-level-meter/blob/main/configs/wdt_include.h

Alternatively you can comment out (or delete) include section and `esp_task_wdt_init` function call in config completely, but sometimes it might cause device restarts in case of long requests (longer than 5s by default) - it could happen in my experience with sensor.community servers. So for better stability I'd recommend to actually leave `esp_task_wdt_init` as is, but add `wdt_include.h` file as I said above.

**Reply by pe0ter** [reply]
So simple, sorry to have bothered you with this
